### DISCUSSION TITLE
Vista OSD works, O3 does not

### DISCUSSION THREAD

**Discussion by dcan999** [iNavFlight/inav-configurator#2030] [General]
Pulling my hair out over this. I've been using my Vista/F405-WMN for over a year without issues, from 6.0.0 through 7.1.0.  I was was using another plane with O3 without issues, but I stopped flying the O3 plane last year. I went to take it out yesterday and noticed I wasn't getting any OSD (INAV or the DJI Voltage reading) so I checked all the usual stuff, rx-tx wires, configuration etc and everything seemed fine. 

I refreshed both the O3 and goggles firmwares and also reflashed 7.1.0 onto the FC, triple checked the config again, still no OSD. I have a brand new O3 unit, so I updated it, plugged it into the FC, STILL no OSD. I then plugged a Vista into the FC and OSD works fine. I plugged a different, new Vista in, and OSD also works fine. I think that ruled out the FC or UART being the issue. I plugged both the Vistas and O3s (one each new and used) into two OTHER FC's and the same situation, OSD works on Vista, doesn't work on O3. 

I am just trying to rule out INAV/my settings being the issue, since all my Vistas work, that would point to a good config, but all my O3's don't work.....so unless my brand new O3 and previously working O3 both somehow broke, I am at a loss as to what's going on. Last I remembered, the Vista and O3 both need the same settings in INAV (OSD slider ON and MSP Displayport selected under peripherals) which is how I have it set. 

I even tried to flash one FC back to 6.0.0, and got the same situation, OSD works on Vista and not on O3.  I also was browsing around the CLI and found the osd_telemetry with OFF, ON and TEST. It was OFF, not knowing what it does (couldn't find anything on here about it) I turned it ON and also to TEST, but in all three settings, got the same results, Vistas work, O3 don't. 

Does anyone see anything I am missing in either my setup/settings of diagnostics?

**Comment by MrD-RC** [comment]
Vista and O3 may use the same settings on the Ports page. If the vista is flashed with WTF.

But on the OSD page the settings are different. The vista will be set to WTF. But the O3 will need to be set to one of the BFCOMPAT OSD types.

**Comment by dcan999** [comment]
Thanks for the reply Darren. I am using the Goggles 2 so the Vista is just flashed with the normal DJI firmware (with ham hack on the goggles). I have the OSD set to BFHDCOMPAT for both vista and O3 since that layout always worked, and it does, for the vistas, still nothing on the O3's. I think the fact that even the DJI OSD voltage is 0.0  means the O3 isn't getting MSP data from the FC, but I can't figure out why. 

I've tried changing the peripheral port to DJI FPV (knowingly wrong) and still no OSD on O3.  

I've checked about 30x now my Rx goes to Tx and vice versa, since that's the most common issue on no OSD on O3, but I definitely have them correct, and just for giggles, I flipped the wires just in case, to rx-rx and tx-tx (obviously wrong) and still no change, no OSD on O3. 

I can't for the life of me think of what else it could be other than the mightily coincidental chance I somehow have a previously working O3 and a brand new O3 both with broken rx/tx ports or something.

**Comment by dcan999** [comment]
I also have always had the auto low power state in the goggles/o3 set to OFF, I just turned it ON (so the O3 is in low power until you arm) and I just noticed it wont leave low power state, meaning the O3 is not getting the ARM command from the FC, further showing the O3 is not getting any MSP data from the FC, but since I've checked my wires a million times and confirmed the UART is OK since the Vistas work fine, I guess that means my O3 are somehow defective?

**Comment by dcan999** [comment]
So the plot thickens, I have a Bee35 with a Speedybee F4 Mini FC stack that I haven't flown yet (waiting for another new O3 unit to arrive), also with 7.1.0 on it. I connected that to one of my O3s and the OSD magically works fine. I am comparing the config from the F4 mini to my 405-WMN's and nothing looks different in regards to the MSP stuff....so what could that mean? I know the 405-WMN UARTs are fine since the Vistas work on them....now I'm even more confused.

**Comment by dcan999** [comment]
I believe this is the part of the diff that would show an issue, both seem to be the same (UART 1 on the Bee35 and UART 3 on the Dolphin)
![diffcomp](https://github.com/iNavFlight/inav-configurator/assets/134864952/06108a20-9aa0-445a-8c82-2fcdb1b203d4)

I really don't think it's the FC's since the vistas work on all of them, but the O3 only work on the F4 mini....so confusing. What am I missing?

**Comment by dcan999** [comment]
Also just tried it on a new F405-Wing V2, and a new H743-WING V3...I even flashed the Bee35 Diff (that the O3 OSD works with on the F4 Mini FC) and it still doesn't work. What the heck? Could this be a Matek specific problem? The speedybee F4 mini is the only non Matek FC I own so I can't really test more in that regard.

**Comment by stemsz** [comment]
Bonjour.
Je rencontre le même problème et sollicite votre aide. Je maitrise bien betaflight et j'ai décidé pour l'un de mes quads de le passer en INAV 7.1.1 le quad vol bien, mais impossible d'avoir le retour vidéo de mon O3 dans mes goggles V2. Bizarrement il n'ai même pas alimenté (ne chauffe pas lorsque la lipo est branchée). avez-vous une idée sur ce problème ?.
Stack MK4 MAMBAF405_2022A.
